update.py

```diff
@@ -219,6 +226,64 @@ def copy_build(site):
         shutil.rmtree(olddir)
 
 
+def copy_and_keep_build(site):
+    """
+    Copies each site into the target location and keep last "n" builts as backups
+    """
+    global error_count
+    for wiki in ALL_WIKIS:
+        if site=='common':
+            continue
+        if not site==None and not site==wiki:
+            continue
+        debug('copy: %s' % wiki)
+        targetdir = os.path.join(args.destdir, wiki)
+
+        debug("Creating  temporary folders")
+        previousdir = os.path.join(args.backupdestdir)
+        debug('Recreating %s' % previousdir )
+        if not os.path.exists(previousdir):
+            os.mkdir(previousdir)
+        olddir = os.path.join(previousdir, str(building_time) )
+        debug('Recreating %s' % olddir )
+        if not os.path.exists(olddir):
+            os.mkdir(olddir)
+
+
+        if os.path.exists(targetdir):
+            debug('Moving %s into %s' % (targetdir,olddir) )
+            shutil.move(targetdir, olddir)
+
+        sourcedir='./%s/build/html/' % wiki
+
+        html_moved_dir = os.path.join(args.destdir, 'html')
+        try:
+            subprocess.check_call(['mv', sourcedir, html_moved_dir])
```

Comment:
Any chance of using `os.rename` here?

Shelling out is awful - and yeah, I know we already do a lot of it in here.

```

pbarker@bluebottle:~$ mkdir pop
pbarker@bluebottle:~$ python
Python 2.7.17 (default, Apr 15 2020, 17:20:14) 
[GCC 7.5.0] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import os
>>> os.rename("pop", "/tmp/pop")
>>> 
pbarker@bluebottle:~$ ls /tmp/pop
pbarker@bluebottle:~$ 
```

Reply:
I do agree that using the shell is not a good idea. As the wiki scripts are being here for a while and me kinda new around, I make some efforts to maintain it as the same as possible (i.e.: maintain the original methods and insert new methods and enable them IFF an argument is used).

For this specific case, I would like to maintain this set of possible exceptions as the original. Do you mind? Otherwise, I would need to check the same collection of possible errors, and I am not aware of them.